### PR TITLE
Fix ByteString literals

### PR DESCRIPTION
**Pull request by vmchale**
This makes it conform to the spec.

### CODE REVIEW

**Comment by mchakravarty** [comment]
BTW, I think, it is worthwhile doing this for the byte strings specifically (despite the existing tests) as this is one of the more complex corners of the lexer.

**Comment by vmchale** [comment]
@mchakravarty Will do! I have yet to add the golden tests, but I will inquire about the error message that should be generated. I think parsing/lexing is a sensible place for it to occur.

File: language-plutus-core/src/Language/PlutusCore/Name.hs [hunk 1]
```diff
@@ -20,14 +20,14 @@ import           Data.Text.Encoding   (decodeUtf8)
 import           PlutusPrelude
 
 -- | A 'Name' represents variables/names in Plutus Core.
-data Name a = Name { attribute :: a
-                   , asString  :: BSL.ByteString -- ^ The identifier name, for use in error messages.
-                   , unique    :: Unique -- ^ A 'Unique' assigned to the name during lexing, allowing for cheap comparisons in the compiler.
+data Name a = Name { nameAttribute :: a
+                   , asString      :: BSL.ByteString -- ^ The identifier name, for use in error messages.
```
Comment (mchakravarty) [inline language-plutus-core/src/Language/PlutusCore/Name.hs]: Why not change `asString` to `nameString` to name all fields consistently?

Comment (vmchale) [reply language-plutus-core/src/Language/PlutusCore/Name.hs]: Will do!

**Comment by mchakravarty** [comment]
@vmchale Thanks for the fixes. I'll merge this now, but we are still missing two sorts of tests:

1. Golden tests triggering overflow in integer literals and in bytestring literals. Alternatively, you could make that Hedgehog tests that generate values that are either a close to the specified size of an integer and byte string, but include literals that just fit into the size as well as literals that are too big [CGP-203].
2. Hedgehog roundtrip tests that generate super simple PLC source code with just an integer or bytestring literal, run it through the parser and pretty-printer to check we get the same source as what we generated [CGP-204].

We need to make sure to test these things well as the security of the blockchain can depend on it. (I made YouTrack tickets, so we don't forget.)

**Pull request merged**